src/qobj/arithmetic_and_attributes.jl

```diff
@@ -576,14 +578,14 @@ function _ptrace_oper(QO::AbstractArray, dims::Union{SVector,MVector}, sel)
 
     ρmat = reshape(QO, reverse(vcat(dims, dims))...)
     topermute = 2 * nd + 1 .- qtrace_sel
-    ρmat = PermutedDimsArray(ρmat, reverse(topermute))
+    ρmat = permutedims(ρmat, reverse(topermute)) # TODO: use PermutedDimsArray when Julia v1.11.0 is released
```

Comment:
why do we have to wait until `Julia v1.11` ?

Reply:
Because it generates type instabilities. They fixed with a PR which will be with their release.

Reply:
okay!